src/main/java/com/labsynch/labseer/service/BulkLoadServiceImpl.java

```diff
@@ -2391,31 +2393,30 @@ private Parent updateParentAliases(Parent parent, Parent foundParent, String che
 		// Note: This is not the most efficient / concise code to do this process
 		// however due to readability and the assumingly small number of aliases 
 		// parents will have this is fine 
-
-		Set<String> parentAliasStrings = new HashSet<>();
+		HashMap<String, ParentAlias> newAliases = new HashMap<String, ParentAlias>();
 		for(ParentAlias alias : parent.getParentAliases()){
 			if (! alias.isIgnored()) {
-				parentAliasStrings.add(alias.getAliasName());
+				newAliases.put(alias.getAliasName(), alias);
 			}
 		}
-		Set<String> foundParentAliasStrings = new HashSet<>(); 
+		HashMap<String, ParentAlias> existingParentAliases = new HashMap<String, ParentAlias>();
 		for(ParentAlias alias : foundParent.getParentAliases()){
 			if (! alias.isIgnored()) {
-				foundParentAliasStrings.add(alias.getAliasName());
+				existingParentAliases.put(alias.getAliasName(), alias);
 			}
 		}
-
-		Set<ParentAlias> unionParentAliases = parent.getParentAliases();
+		// Create a set to hold the union of all unique aliases
+		// Start with the existing aliases
+		Set<ParentAlias> unionParentAliases = foundParent.getParentAliases();
+		// Find any new aliases not already in the old set, and add them
+		Set<String> newAliasStrings = newAliases.keySet().stream().filter(e -> 
+			!existingParentAliases.keySet().contains(e)).collect(Collectors.toSet());
+		for(String newAliasString : newAliasStrings) {
+			unionParentAliases.add(newAliases.get(newAliasString));
+		}
 		// Parent Would Be Found Parent w/ Appropriate Updates
 		parent = foundParent;
 		// Update Parent Object to Have "Union" of All Aliases 
-		for(ParentAlias oldAlias : parent.getParentAliases()) {
-			// If oldAlias Not In UnionParentAliases (Use Previously Set<str> to Do Compare)
-				// Add to Aliases Union List 
-			if(! oldAlias.isIgnored() && !parentAliasStrings.contains(oldAlias.getAliasName())) {
-				unionParentAliases.add(oldAlias);
-			}
-		}
 		parent.setParentAliases(unionParentAliases);
```

Comment:
@dalejerikson 
The result is that we'll have:
- old non-overlap ParentAliases
- new non-overlap ParentAliases
- overlapping ones represented by the new Java object (i.e. with no `id`)

The old overlapping aliases are not included in this set, but they still exist in the database. They don't get ignored or deleted just by saving the `parent` with a set of `parentAliases` that doesn't include them.

The result is that any overlapping aliases get duplicated. We save the new copies of them as new rows in `parent_alias` and the old copies of them still exist in `parent_alias`.

Reply:
The main difference between my implementation and yours is that I preferentially use the *old* ParentAlias objects and only add "new non-overlap ParentAliases"